app/models/post.rb

```diff
@@ -4,4 +4,15 @@ class Post < ApplicationRecord
 
   belongs_to :day
   belongs_to :user
+
+  def self.with_matching_group(user_group_id)
+    all_posts = joins(:day).select('posts.*, days.group_id as group')
+    all_posts.find_all {|post| post.group == user_group_id}
+  end
+
+  def self.matching_group_and_user(user)
+    all_posts = joins(:day).select('posts.*, days.group_id as group')
+    all_posts.find_all {|post| post.group == user.group.id && post.user.id == user.id}
+  end
```

Comment:
This method does the same as the above but also filters for a matching user. The same questions above apply here as far as refactoring and the group issue.